src/dmd/access.d

```diff
@@ -543,7 +543,7 @@ extern (C++) bool checkSymbolAccess(Scope *sc, Dsymbol s)
  * but doesn't recurse nor resolve aliases because protection/visibility is an
  * attribute of the alias not the aliasee.
  */
-public Dsymbol mostVisibleOverload(Dsymbol s)
+public Dsymbol mostVisibleOverload(Dsymbol s, Module mod = null)
```

Comment:
You can get rid of the optional parameter and obtain the module form s by calling `s.getModule` [1]

[1] https://github.com/dlang/dmd/blob/master/src/dmd/dsymbol.d#L327

Reply:
No? The module is the module the symbol is supposed to be visible *from*.